handler.c

```diff
@@ -154,6 +167,7 @@ struct efi_variable *var_list;
 bool secure_boot_enable;
 bool auth_enforce = true;
 bool persistent = true;
+bool ppi_locked = false;
```

Comment:
You don't need to store this state separately since the `TCG2_PHYSICAL_PRESENCEFLAGSLOCK_NAME` variable already holds the state.

That way, there's no chance them getting out of sync, and you don't need the restore_ppi_lock() function to deal with migration.

You would just need to do a `internal_get_variable()` call in `check_ppi_variables()` line 1324.

Reply:
All this checking of uefi variables seemed very heavy weight - so my natrual reaction is to cache the result for easy access. 
But I guess there is no need for this to be efficent.